modules/oe_subscriptions_anonymous/src/Form/AnonymousSubscribeForm.php

```diff
@@ -194,23 +189,25 @@ public function submitForm(array &$form, FormStateInterface $form_state) {
       // defined.
       $form_state->setRedirectUrl(Url::fromRoute('<front>'));
     }
+    // Storing redirect disabled value to use in AjaxSubmit.
+    if ($this->isAjax()) {
+      $form['redirect_disabled'] = $form_state->isRedirectDisabled();
+    }
   }
 
   /**
    * {@inheritdoc}
    */
   protected function successfulAjaxSubmit(array $form, FormStateInterface $form_state) {
+    // Adding redirection from regular form submit variables. By default,
+    // Drupal\Core\Form\FormBuilder disables the redirection in AJAX submission.
+    // This is byassed in order to allow the message template from theme take
+    // place. Drupal\Core\Ajax\MessageCommand does not use theme template.
+    $form_state->disableRedirect($form_state->get('redirect_disabled'));
     $response = new AjaxResponse();
+    $url = new RedirectCommand($form_state->getRedirect()->toString());
```

Comment:
This is not really what I meant with "store the value". First, we should never use the form build array to store data like that. In general, what I meant was:
```
$is_redirect_disabled = $form_state->isRedirectDisabled();
$form_state-> disableRedirect(FALSE);
$redirect = $form_state->getRedirect();
$form_state->disableRedirect($is_redirect_disabled);
```

Regarding the comment, I wanted a comment on why we are forced to disable the redirect. The comment about replacing the ajax message with a reload is already in the test, here we need to explain why we cannot retrieve the redirect from the form state without changing the form state value.

Reply:
Fixed and changed